python/cudf/cudf/_lib/io/utils.pyx

```diff
@@ -42,10 +43,16 @@ cdef source_info make_source_info(src) except*:
 
 # Converts the Python sink input to libcudf++ IO sink_info.
 cdef sink_info make_sink_info(src, unique_ptr[data_sink] * data) except*:
-    if isinstance(src, io.IOBase):
+    if isinstance(src, io.TextIOBase):
+        if codecs.lookup(src.encoding).name != "utf-8":
+            raise NotImplementedError(f"Unsupported encoding f{src.encoding}")
+        data.reset(new iobase_data_sink(src.buffer))
+        return sink_info(data.get())
+    elif isinstance(src, io.IOBase):
         data.reset(new iobase_data_sink(src))
         return sink_info(data.get())
     elif isinstance(src, (int, float, complex, basestring, os.PathLike)):
```

Comment:
Why is the source allowed to be `int`, `float`, or `complex` here?
Is the `os.path.expanduser()` line I added below ok if src is `os.PathLike`?

Reply:
I copied the int/float/complex logic to be consistent with make_source_info, where it was added @ https://github.com/rapidsai/cudf/commit/87f7f25d99 . Reading the comments there it seems like the intent was to support passing a numeric file descriptor - but now I'm not sure if that will work (I suspect that converting to a string when passing to source_info will cause it to try to open a filename rather than the FD).  @trxcllnt whats your take on this?

Reply:
If supporting file descriptors is the intention, I suppose we could add support for that on the C++ side. It doesn't look like this could be added to the existing `file_sink` as it uses fstream which apparently doesn't support file descriptors, but we could add another `data_sink` implementation that uses `fdopen()` instead.

Reply:
Yeah I think that's right. I can't remember now why I did this... maybe there was a test that used a file descriptor before and I carried this over by accident? Either way, it does seem like casting to a string would make C++ think this is a filename instead of an fd.

Reply:
In that case -  I think we should change the checks in make_source_info and make_sink_info to ``` isinstance(src, (basestring, os.PathLike))```

Reply:
Ok. I cleaned up the `expanduser()` call as well since it seems to work with PathLike inputs and returns a str.